pkg/stub/handler.go

```diff
@@ -2,96 +2,410 @@ package stub
+		return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to delete before create dockerconfig secret the openshift namespace: %v")
+	}
+	if deleted {
+		logrus.Printf("registry dockerconfig secret %s was deleted", h.registrySecret.Name)
+		h.registrySecret = nil
+		h.operatorCfg.Status.SecretCreated = false
+		h.sdkwrapper.Update()
+	} else {
+		secretToCreate := corev1.Secret{}
+		h.registrySecret.DeepCopyInto(&secretToCreate)
+		secretToCreate.Namespace = ""
+		secretToCreate.ResourceVersion = ""
+
+		_, err = h.secreteclientwrapper.Create("openshift", &secretToCreate)
+		if err != nil {
+			return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to create registry dockerconfig secret in openshift namespace : %v")
+		}
+		logrus.Printf("created/updated dockerconfig secret %s in openshift namespace", h.operatorCfg.Spec.RegistryCredentials.Name)
+
+		h.operatorCfg.Status.SecretCreated = true
+		h.sdkwrapper.Update()
+
+	}
+
+	status := v1alpha1.ResourceCondition{
+		Type: v1alpha1.SecretUpdateComplete,
+		Time: kapis.Now(),
+	}
+	h.operatorCfg.Status.Conditions = append(h.operatorCfg.Status.Conditions, status)
+	err = h.sdkwrapper.Update()
+	if err != nil {
+		if !kerrors.IsConflict(err) {
+			return h.processError(v1alpha1.SamplesUpdateFailed, err, "failed adding success condition to config: %v")
+		}
+		logrus.Println("got conflict error on success status update, just going to forgo this time vs. retry")
+	}
+
+	return nil
 }
 
 func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
-	switch o := event.Object.(type) {
+	switch event.Object.(type) {
+	case *corev1.Secret:
+		dockercfgSecret, _ := event.Object.(*corev1.Secret)
+
+		if h.operatorCfg == nil {
+			h.registrySecret = dockercfgSecret
+			return nil
+		}
+
+		if h.operatorCfg.Spec.RegistryCredentials.Name == dockercfgSecret.Name {
+			if h.registrySecret != nil &&
+				h.registrySecret.ResourceVersion == dockercfgSecret.ResourceVersion &&
+				h.operatorCfg.Status.SecretCreated &&
+				!event.Deleted {
+				return nil
+			}
+
+			h.registrySecret = dockercfgSecret
+
+			err := h.manageDockerCfgSecret(event.Deleted)
+			if err != nil {
+				return err
+			}
+		}
+
 	case *v1alpha1.SamplesResource:
-		logrus.Printf("ctrl cfg %#v", o)
-		restconfig, err := getRestConfig()
-		if err != nil {
-			logrus.Errorf("failed to get rest config : %v", err)
-			return err
+		if event.Deleted {
+			h.operatorCfg = nil
+			return nil
 		}
-		tempclient, err := getTemplateClient(restconfig)
-		if err != nil {
-			logrus.Errorf("failed to get template client : %v", err)
-			return err
+
+		opcfg, _ := event.Object.(*v1alpha1.SamplesResource)
+
+		if h.operatorCfg != nil && h.operatorCfg.ResourceVersion == opcfg.ResourceVersion {
+			return nil
+		}
+		h.operatorCfg = opcfg
+
+		if len(h.operatorCfg.Spec.RegistryCredentials.Name) > 0 &&
+			h.registrySecret != nil &&
+			!h.operatorCfg.Status.SecretCreated {
+			h.manageDockerCfgSecret(false)
```

Comment:
is this needed at all?  Since you're watching for the creation of the secret, can't you just handle it there?

Reply:
Yes this is needed if a Secret event arrives before the initial SamplesResource event

Reply:
ah, good point, we don't want the import to fail due to the missing secret since nothing will automatically retry the import.  (We probably need to mark our imagestreams for scheduled imports at some point, but that's a side thread).

Reply:
I was not suggesting we start marking imagestreams for scheduled import for now, that's a change that's going to have far reaching impacts on users.

Allowing them to override the samples to allow insecure import is a good point...I don't think ansible install handles that today, but it would be a nice to have here.  I wouldn't deem it critical (i'd really prefer to avoid doing much mutation of the imagestreams if we can help it, beyond the registry replacement you're already doing)